File: src/rails/game/model/BonusModel.java
package rails.game.model;

import rails.game.Bonus;
import rails.game.Currency;
import rails.game.RailsItem;
import rails.game.state.ArrayListState;
import rails.game.state.Model;

public class BonusModel extends RailsModel {
    
    private ArrayListState<Bonus> bonuses;
    
    protected BonusModel(RailsItem parent, String id) {
        super(parent, id);
    }

    public static BonusModel create(RailsItem parent, String id){
        return new BonusModel(parent, id);
    }
    
    @Override
    public RailsItem getParent() {
        return (RailsItem)super.getParent();
    }

    public void setBonuses(ArrayListState<Bonus> bonuses) {
        this.bonuses = bonuses;
        bonuses.addModel(this);
    }

    @Override
    public String toText() {
        if (bonuses == null || bonuses.isEmpty()) return "";

        StringBuffer b = new StringBuffer("<html><center>");

        for (Bonus bonus : bonuses.view()) {
            if (b.length() > 14) {
                b.append("<br>");
            }
            b.append(bonus.getIdForView()).append("+").append(Currency.format(getParent(), bonus.getValue()));
        }

        return b.toString();
    }
}

File: dotnet/RailsLib.Net/Net/Game/Model/BonusModel.cs
using GameLib.Net.Game.Financial;
using GameLib.Net.Game.State;
using System;
using System.Collections.Generic;
using System.Text;

namespace GameLib.Net.Game.Model
{
    public class BonusModel : RailsModel
    {
        private ListState<Bonus> bonuses;

        protected BonusModel(IRailsItem parent, string id) : base(parent, id)
        {

        }

        public static BonusModel Create(IRailsItem parent, string id)
        {
            return new BonusModel(parent, id);
        }

        //public IRailsItem getParent()
        //{
        //    return (RailsItem)super.getParent();
        //}

        public void SetBonuses(ListState<Bonus> bonuses)
        {
            this.bonuses = bonuses;
            bonuses.AddModel(this);
        }

        override public string ToText()
        {
            if (bonuses == null || bonuses.IsEmpty) return "";

            StringBuilder b = new StringBuilder("<html><center>");

            foreach (Bonus bonus in bonuses.View())
            {
                if (b.Length > 14)
                {
                    b.Append("<br>");
                }
                b.Append(bonus.GetIdForView()).Append("+").Append(Bank.Format(Parent, bonus.Value));
            }

            return b.ToString();
        }
    }
}
